### ISSUE TITLE
Add "ping" and "pong" keywords 

### ISSUE THREAD

**Issue by snk-js** [TodePond/DreamBerd#177]
This will enable compiler to read the code back and forth many times between the keyword ping and pong:

```
ping 3!
print('hello')
print('world')
print('-')
pong!

// hello  <- first ping
// world
// - <- first pong
// world
// hello <- second ping
// world
// - <- second pong
// world
// hello <-third ping, ends ping pong.
```

**Comment by stohrendorf** [comment]
Isn't this a loop? DremBerd has no loops.

**Comment by snk-js** [comment]
> Isn't this a loop? DremBerd has no loops.

makes sense, in my opinion a classical loop, but this is a ping pong, the code don't repeat it reverses reading as soon it gets to pong and re-reverse as soon reaches ping.

**Comment by TodePond** [comment]
technically it's not a loop. It's repetition, which is fine

**Comment by arily** [comment]
I think we need to find a way to prevent devs using it like a loop
```js
const const arr = ['maybe', 'you', 'invented','a','loop

const var now = 0!
ping arr.length!
  now += 1
  print(arr[now
  print(" "
pong!
```

could be solved if codes between ping and pong will have an isolated scope (anything outside ping pong is not accessible)

```js
const const arr = ['maybe', 'you', 'invented','a','loop

ping arr.length!
  print(arr.push( // Compiler Error: unknown identifier: arr
  print(" " // Compiler Error: unknown identifier: print
pong!
```

**Comment by snk-js** [comment]
@arily  I have requested a feature that may have a good match with ping pong too:
#178 it create portals to go to another line of code.

if we join ping pong using portals inside of it we have almost a perfect usage of repetition and scope access without the concept of loops:

```
const const arr = ['maybe', 'you', 'invented','a','loop

let i = 0;

portal A:
arr.push(arr[i])!
go B

ping arr.length!
  portal B:
  go A
pong!
```

inside of ping, go A will read lines with ```arr.push``` and then will go to portal B (defined inside ping pong) to repeat again the passage of portal A

**Comment by arily** [comment]
@snk-js 
Did we just solved borrow checker and locks entirely? Can't wait to see how performant it is. Gonna be so much faster than javascript not to mention Typescript I guess.

**Comment by snk-js** [comment]
> @snk-js Did we just solved borrow checker and locks entirely? Can't wait to see how performant it is. Gonna be so much faster than javascript not to mention Typescript I guess.

we might eliminate borrow check and locks but we also create paradox opportunities, as we dealing with portals literally, and to decrease paradox we might also create `close` to  close portals before a paradox be created
